File: ejemplos/03-peticion-respuesta/servidor.py
import time
from nerve import NexusClient

class ServidorCalculadora:
    def __init__(self):
        self.cliente = NexusClient()

    def atender_peticion(self, mensaje_crudo):
        remitente = mensaje_crudo.get("from")
        payload = mensaje_crudo.get("payload", {})
        
        accion = payload.get("accion")
        
        # Validamos que el mensaje es para nosotros y es una acción conocida
        if accion == "sumar":
            a = payload.get("a", 0)
            b = payload.get("b", 0)
            resultado = a + b
            
            print(f"🧮 Solicitud de suma recibida de '{remitente}': {a} + {b} = {resultado}")
            
            # Construimos la respuesta
            respuesta = {
                "accion": "respuesta_operacion",
                "resultado": resultado
            }
            
            # Mandamos la respuesta de vuelta a quien nos lo pidió
            print(f"Enviando resultado a '{remitente}'...")
            self.cliente.send(remitente, respuesta)

    def iniciar(self):
        self.cliente.connect("servidor_calculadora")
        print("✅ Servidor Calculadora en línea y esperando peticiones...")
        
        self.cliente.listen(self.atender_peticion)
        
        try:
            while True:
                time.sleep(1)
        except KeyboardInterrupt:
            print("\nApagando servidor...")
            self.cliente.disconnect()

if __name__ == "__main__":
    app = ServidorCalculadora()
    app.iniciar()

File: archivo-historico/ejemplos/nerve/03-peticion-respuesta/servidor.go
package main

import (
	"fmt"
	"os"
	"os/signal"
	"syscall"

	"nerve"
)

type ServidorCalculadora struct {
	cliente *nerve.NexusClient
}

func (s *ServidorCalculadora) atenderPeticion(mensajeCrudo nerve.Message) {
	remitente := mensajeCrudo.From
	payload, ok := mensajeCrudo.Payload.(map[string]interface{})
	if !ok {
		return
	}

	accion, ok := payload["accion"].(string)
	if ok && accion == "sumar" {
		a, _ := payload["a"].(float64)
		b, _ := payload["b"].(float64)
		resultado := a + b

		fmt.Printf(" Solicitud de suma recibida de '%s': %v + %v = %v\n", remitente, a, b, resultado)

		respuesta := map[string]interface{}{
			"accion":    "respuesta_operacion",
			"resultado": resultado,
		}

		fmt.Printf("Enviando resultado a '%s'...\n", remitente)
		s.cliente.Send(remitente, respuesta)
	}
}

func (s *ServidorCalculadora) iniciar() {
	s.cliente.Connect("servidor_calculadora")
	fmt.Println("✅ Servidor Calculadora en línea y esperando peticiones...")

	s.cliente.Listen(s.atenderPeticion)

	c := make(chan os.Signal, 1)
	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
	<-c

	fmt.Println("\nApagando servidor...")
	s.cliente.Disconnect()
}

func main() {
	app := &ServidorCalculadora{
		cliente: nerve.NewNexusClient(),
	}
	app.iniciar()
}
